packages/console/src/ConsoleStatusBar.tsx

```diff
@@ -1,136 +1,110 @@
-import { DropdownAction, Tooltip } from '@deephaven/components';
-import { CanceledPromiseError, Pending } from '@deephaven/utils';
-import ConsoleMenu from './ConsoleMenu';
+import {
+  Button,
+  DropdownActions,
+  DropdownMenu,
+  PopperOptions,
+  Tooltip,
+} from '@deephaven/components';
 import './ConsoleStatusBar.scss';
 
+const POPPER_OPTIONS: PopperOptions = { placement: 'bottom-end' };
+
 interface ConsoleStatusBarProps {
-  children: ReactNode;
+  children?: ReactNode;
   dh: typeof DhType;
   session: DhType.IdeSession;
-  openObject: (object: DhType.ide.VariableDefinition) => void;
-  objects: DhType.ide.VariableDefinition[];
-  overflowActions: () => DropdownAction[];
-}
-
-interface ConsoleStatusBarState {
-  isDisconnected: boolean;
-  isCommandRunning: boolean;
+  overflowActions?: DropdownActions;
 }
 
-export class ConsoleStatusBar extends PureComponent<
-  ConsoleStatusBarProps,
-  ConsoleStatusBarState
-> {
-  static defaultProps = {
-    children: null,
-  };
-
-  constructor(props: ConsoleStatusBarProps) {
-    super(props);
-
-    this.handleCommandStarted = this.handleCommandStarted.bind(this);
-    this.handleCommandCompleted = this.handleCommandCompleted.bind(this);
-
-    this.pending = new Pending();
-
-    this.state = {
-      isDisconnected: false,
-      isCommandRunning: false,
-    };
-  }
-
-  componentDidMount(): void {
-    this.startListening();
-  }
+function ConsoleStatusBar({
+  children,
+  dh,
+  session,
+  overflowActions,
+}: ConsoleStatusBarProps): ReactElement {
+  const [isCommandRunning, setIsCommandRunning] = useState(false);
 
-  componentWillUnmount(): void {
-    this.stopListening();
-    this.cancelPendingPromises();
-  }
-
-  pending: Pending;
-
-  startListening(): void {
-    const { dh, session } = this.props;
-    session.addEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
+  const handleCommandStarted = useCallback(async (event: CustomEvent) => {
+    setIsCommandRunning(true);
 
-  stopListening(): void {
-    const { dh, session } = this.props;
-    session.removeEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
-
-  cancelPendingPromises(): void {
-    this.pending.cancel();
-  }
-
-  handleCommandStarted(event: CustomEvent): void {
     const { result } = event.detail;
-
-    this.pending
-      .add(result)
-      .then(() => this.handleCommandCompleted(null))
-      .catch(error => this.handleCommandCompleted(error));
-
-    this.setState({
-      isCommandRunning: true,
-    });
-  }
-
-  handleCommandCompleted(error: unknown): void {
-    // Don't update state if the promise was canceled
-    if (!(error instanceof CanceledPromiseError)) {
-      this.setState({
-        isCommandRunning: this.pending.pending.length > 0,
-      });
+    try {
+      await result;
+    } catch (error) {
+      // No-op, fall through
```

Comment:
Why the change to the logic here? Previously looks like we did something w/ the error

Reply:
Before we just used the error to check the pending count. This just cleans up the logic.
I did change it to use a count instead of a flag in case of multiple commands pending.